### PR TITLE
optimize case where div denominator is -1

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112878

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/codegen/common.py [hunk 1]
```diff
@@ -349,6 +349,10 @@ def _print_ModularIndexing(self, expr):
 
     def _print_FloorDiv(self, expr):
         x, div = expr.args
+        if div == -1:
+            return f"{self.paren(self.doprint(x * -1))}"
+        elif div == 1:
+            return f"{self.paren(self.doprint(x))}"
```
Comment (peterbell10) [inline torch/_inductor/codegen/common.py:355]: Is this condition already covered here?
https://github.com/pytorch/pytorch/blob/2c3ab605068b7efc09f63c62048671750f69d6da/torch/utils/_sympy/functions.py#L65-L68

File: torch/_inductor/codegen/common.py [hunk 2]
```diff
@@ -349,6 +349,10 @@ def _print_ModularIndexing(self, expr):
 
     def _print_FloorDiv(self, expr):
         x, div = expr.args
+        if div == -1:
+            return f"{self.paren(self.doprint(x * -1))}"
```
Comment (peterbell10) [inline torch/_inductor/codegen/common.py:353]: Can we handle this in `FloorDiv.eval`? That would potentially allow SymPy to make further simplifications.

**Review by peterbell10 — COMMENTED** [review]
The simplification makes sense, but I'm not sure the printer is the right place for this.

File: torch/_inductor/codegen/common.py (hunk 2 above)
Comment (guilhermeleobas) [reply torch/_inductor/codegen/common.py]: Moved.

File: torch/_inductor/codegen/common.py (hunk 1 above)
Comment (guilhermeleobas) [reply torch/_inductor/codegen/common.py]: Yes, it is. I've updated the code to remove the duplicated condition.

File: torch/utils/_sympy/functions.py [hunk 3]
```diff
@@ -66,6 +66,8 @@ def check_supported_type(x):
             return base
         if base.is_real and divisor == 1:
             return sympy.floor(base)
+        if divisor == -1:
```
Comment (peterbell10) [inline torch/utils/_sympy/functions.py:69]:
```suggestion
        if base.is_integer and divisor == -1:
```

File: torch/_inductor/codegen/cpp.py [hunk 4]
```diff
@@ -320,6 +320,8 @@ def _print_ModularIndexing(self, expr):
 
     def _print_FloorDiv(self, expr):
         x, div = expr.args
+        if div == -1:
+            return f"{self.paren(self.doprint(x * -1))}"
```
Comment (peterbell10) [inline torch/_inductor/codegen/cpp.py:324]: Are we still hitting this case?

Comment (guilhermeleobas) [reply torch/_inductor/codegen/cpp.py:324]: No! Forgot to push this change

File: torch/utils/_sympy/functions.py [hunk 5]
```diff
@@ -66,6 +66,8 @@ def check_supported_type(x):
             return base
         if base.is_real and divisor == 1:
             return sympy.floor(base)
+        if base.is_integer and divisor == -1:
```
Comment (lezcano) [inline torch/utils/_sympy/functions.py:69]: If it's float we can make it into a floor op I guess

Comment (lezcano) [reply torch/utils/_sympy/functions.py:69]: We can even merge it with the previous if perhaps.

**Review by lezcano — APPROVED** [review]
We can have a small one as well, but otherwise LGTM

**Pull request closed without merging**